### PR TITLE
Improve the user interface of WANDPowderReduction

### PR DESCRIPTION
**Pull request by KedoKudo**
**Description of work.**

- Related Gitlab user story: [PD129](https://code.ornl.gov/sns-hfir-scse/diffraction/powder/powder-diffraction/-/issues/129)
- This is #31154  into `master`

This PR is mostly a user interface tune-up for the current `WANDPowderReduction`:

- User can again select directly from ADS from GUI
- The candidate items for `Target` property is reduced to the three that are relevant to HB2C (Theta, ElasticQ, ElasticDSpacing)
- Change `EFixed` to wavelength with a default value of 1.4865A
- The default bin number is set to `1000` per request

**To test:**
Since this is a GUI tune-up, the best way to test is to build the branch and launch the GUI to check.
The functionality check can be done via existing unit test.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/WANDPowderReduction.py [hunk 1]
```diff
@@ -117,9 +117,28 @@ def PyInit(self):
             doc="The mask from this workspace will be applied before reduction",
         )
 
-        self.copyProperties("ConvertSpectrumAxis", ["Target", "EFixed"])
+        self.declareProperty(
+            "EFixed",
+            37.02,
```
Comment (rosswhitfield) [inline Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/WANDPowderReduction.py:122]: Having a default value for this here seems strange, it is set correctly from wavelength during loading with LoadWAND. If we are doing this then we should remove https://github.com/mantidproject/mantid/blob/f643296c46e4e95bcc4e747612842c7b2e37c0f4/Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/LoadWAND.py#L91 as it will not be used.

Actually, we should remove the [Wavelength](https://github.com/mantidproject/mantid/blob/f643296c46e4e95bcc4e747612842c7b2e37c0f4/Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/LoadWAND.py#L29) from LoadWAND completely as this won't be used and could add to the confusion for the users.

I'm also surprised that they didn't want the ability to set the wavelength instead of energy.

**Review by rosswhitfield — CHANGES_REQUESTED** [review]
The usage examples no longer work https://docs.mantidproject.org/nightly/algorithms/WANDPowderReduction-v1.html so please update them.

File: Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/WANDPowderReduction.py [hunk 2]
```diff
@@ -135,7 +154,8 @@ def PyInit(self):
         )
 
         self.declareProperty(
-            WorkspaceProperty("OutputWorkspace", "", direction=Direction.Output),
+            "OutputWS",
```
Comment (rosswhitfield) [inline Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/WANDPowderReduction.py:157]: Why was this changed from `OutputWorkspace` to `OutputWS`?

File: Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/WANDPowderReduction.py (hunk 1 above)
Comment (rosswhitfield) [reply Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/WANDPowderReduction.py:122]: This always overrides the wavelength value set by LoadWAND, therefore we should remove the option in LoadWAND.

Comment (rosswhitfield) [reply Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/WANDPowderReduction.py:122]: All the usage examples fail because you removed the OutputWorkspace property.

Comment (KedoKudo) [reply Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/WANDPowderReduction.py:122]: Ah, I see.  I will update the usage example now.

As for the removing `wavelength` in `LoadWAND`, I don't think we should do it here since it is not requested by the beamline.

Comment (rosswhitfield) [reply Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/WANDPowderReduction.py:122]: > As for the removing `wavelength` in `LoadWAND`, I don't think we should do it here since it is not requested by the beamline.

I'm requesting it! By making EFixed no longer optionally it's removed the functionally of the Wavelength property in LoadWAND.

**Review by rosswhitfield — APPROVED** [review]
